### ISSUE TITLE
Consider adding gin_login

### ISSUE THREAD

**Issue by rpkoller** [phenaproxima/starshot-prototype#86]
Since gin and gin_toolbar have already being added to Starshot it would make sense for consistency reasons alone to add gin_login to the mix as well (https://www.drupal.org/project/gin_login). It is simply adding a nice touch to the login experience and integrates well into the overall look and feel of gin.

**Comment by phenaproxima** [comment]
I'm not necessarily against adding this, but from a very quick skim of the project page, this looks mainly cosmetic, as opposed to Gin and Gin Toolbar, which make the admin backend more _usable_.

If Gin Login actually improves the convenience and usability of the login page, I'm all for it. But if it's a facelift -- albeit a welcome one -- I'd prefer to add it in some secondary recipe, rather than have it be part of Starshot's OOTB experience.

**Comment by rpkoller** [comment]
good point i agree. and it is mainly a facelift making the styling of the login in line with gin. my main motivation here was for the sake of consistency (i always strive for that) . but i've taken a brief look and it looks like it also adds functionality for headless: 

```
Headless ready
If you're using Drupal as a headless content system, then you can set up the Front page of Drupal that it always shows you the Login screen (if not logged in) or the Content Overview if you're logged in.

Just set in /admin/config/system/site-information the default front page to /user/login?destination=/admin/content
```

**Comment by phenaproxima** [comment]
Given all of that, I think that gin_login might make more sense in the context of a recipe that sets up a headless site. That'd be a perfect use case, IMHO.

I do take your point about consistency. Also, Gin Login frankly does look a lot nicer. On the other hand, it's just a login form and I doubt most users of Drupal will put much stock in how it looks. :) So I think where I land on this is still "I would personally not add it right now, but if other folks really want it, I don't see the harm".

**Comment by gitressa** [comment]
I would also prefer to keep the number of modules from ballooning, if possible. Like Ted Bowman once said, "there's a module for that, don't use it".

**Comment by roromedia** [comment]
I am not a big fan of Gin Login as it uses a ThemeNegotiator to dynamically switch to the admin-theme instead of the frontend theme for user pages (user.login, user.pass and user.register routes) - this makes it harder to get a streamlined theming experience for the whole frontend. A new user could be confused why the backend theme comes in here for theming.
The look and feel of the Gin Login module is very pretty but could be easily rebuilt by using some simple page--user.html.twig (and corresponding page--user--login.html.twig, page--user--password.html.twig and page--user--register.html.twig) templates and some CSS applied to it.

**Comment by pameeela** [comment]
I can see the logic in using the admin theme for the login, I think this matches what many users expect just as far as login = back end. But if you didn't have permissions that had you using the admin theme, it would be confusing for sure.

**Comment by roromedia** [comment]
I see your point but I still feel that a login page should look like a front-end page especially when used in an eCommerce setting like Drupal Commerce where clients should have a seamless store experience and no backend login page in-between.
